src/Symfony/Component/Mailer/Envelope.php

```diff
@@ -85,4 +81,32 @@ public function getRecipients(): array
     {
         return $this->recipients;
     }
+
+    /**
+
+      * Returns true if any address' localpart contains at least one
+      * non-ASCII character, and false if all addresses have all-ASCII
+      * localparts.
+      *
+      * This helps to decide whether to the SMTPUTF8 extensions (RFC
+      * 6530 and following) for any given message.
+      *
+      * The SMTPUTF8 extension is strictly required if any address
+      * contains a non-ASCII character in its localpart. If non-ASCII
+      * is only used in domains (e.g. horst@freiherr-von-mühlhausen.de)
+      * then it is possible to to send the message using IDN encoding
+      * instead of SMTPUTF8. The most common software will display the
+      * message as intended.
+      *
+      * @return bool
```

Comment:
```suggestion
```

Reply:
Not a problem, but could you elaborate on how the newest and best Symfony code uses such syntax?

Reply:
We don't use phpdoc tags for `@param` or `@return` if they don't provide any additional information compared to the native function signature (avoiding to maintain duplicate information).

In practice, this means we'll use them only in 4 cases:

- generic types
- array types (where the phpdoc can provide details about the types _inside_ the array, which is kinda related to generic types)
- callable types (where the phpdoc can document the callable signature)
- cases where the native type will change only in the next major version for BC reasons.

Reply:
Got it. Makes sense. And refreshing to come across someone who has such a considered attitude to documentation tooling.